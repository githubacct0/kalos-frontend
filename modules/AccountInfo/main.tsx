import React, { ChangeEvent } from 'react';
import { User, UserClient } from '@kalos-core/kalos-rpc/User';
import PermIdentityIcon from '@material-ui/icons/PermIdentity';
import EditIcon from '@material-ui/icons/Edit';
import SendIcon from '@material-ui/icons/Send';
import {
  TextField,
  Grid,
  Modal,
  Button,
  Paper,
  Switch,
  Typography,
  Divider,
  IconButton,
  FormControlLabel,
  CssBaseline,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/CloseSharp';

interface props {
  userId: number;
}

interface state {
  user: User.AsObject;
  isModalOpen: boolean;
  isEditing: boolean;
  isLoginModalOpen: boolean;
  searchString: string;
}

export class AccountInfo extends React.PureComponent<props, state> {
  UserClient: UserClient;
  oldPassword: React.RefObject<HTMLInputElement>;
  newPassword: React.RefObject<HTMLInputElement>;
  reTypePassword: React.RefObject<HTMLInputElement>;
  oldLogin: React.RefObject<HTMLInputElement>;
  newLogin: React.RefObject<HTMLInputElement>;
  reTypeLogin: React.RefObject<HTMLInputElement>;

  constructor(props: props) {
    super(props);
    this.state = {
      user: new User().toObject(),
      isModalOpen: false,
      isEditing: false,
      isLoginModalOpen: false,
      searchString: '',
    };
    this.handleUpdateLogin = this.handleUpdateLogin.bind(this);
    this.toggleLoginModal = this.toggleLoginModal.bind(this);
    this.toggleEditing = this.toggleEditing.bind(this);
    this.toggleModal = this.toggleModal.bind(this);
    this.handleUpdatePassword = this.handleUpdatePassword.bind(this);
    this.UserClient = new UserClient('https://core-dev.kalosflorida.com:8443');
    this.oldPassword = React.createRef();
    this.newPassword = React.createRef();
    this.reTypePassword = React.createRef();
    this.oldLogin = React.createRef();
    this.newLogin = React.createRef();
    this.reTypeLogin = React.createRef();
  }

  async fetchUser() {
    const req = new User();
    req.setId(this.props.userId);

    const result = await this.UserClient.Get(req);
    this.setState({
      user: result,
    });
  }

  toggleModal() {
    this.setState(prevState => ({
      isModalOpen: !prevState.isModalOpen,
    }));
  }

  toggleLoginModal() {
    this.setState(prevState => ({
      isLoginModalOpen: !prevState.isLoginModalOpen,
    }));
  }

  async handleUpdateLogin() {
    try {
      if (
        this.oldLogin.current &&
        this.newLogin.current &&
        this.reTypeLogin.current
      ) {
        const oldLogin = this.oldLogin.current.value;
        const newLogin = this.newLogin.current.value;
        const reTypeLogin = this.reTypeLogin.current.value;

        if (newLogin !== reTypeLogin) {
          throw 'Usernames do not match';
        }
        if (oldLogin !== this.state.user.login) {
          throw 'Old Login is incorrect';
        }
        const login = this.oldLogin.current.value;
        const req = new User();
        req.setLogin(newLogin);
        req.setId(this.state.user.id);
        req.setFieldMaskList(['Login']);
        const updatedUser = await this.UserClient.Update(req);
        this.setState({ user: updatedUser });

        this.toggleLoginModal();
      }
    } catch (err) {
      alert(err);
    }
  }

  async handleUpdatePassword() {
    try {
      if (
        this.oldPassword.current &&
        this.newPassword.current &&
        this.reTypePassword.current
      ) {
        const oldPassword = this.oldPassword.current.value;
        const newPassword = this.newPassword.current.value;
        const reTypePassword = this.reTypePassword.current.value;
        if (newPassword.length < 8) {
          throw 'Password needs to be 8 characters long';
        }
        if (newPassword !== reTypePassword) {
          throw 'Passwords do not match';
        }
        if (oldPassword !== this.state.user.pwd) {
          throw 'Old Password is incorrect';
        }
        const req = new User();
        req.setPwd(newPassword);
        req.setFieldMaskList(['Pwd']);
        req.setId(this.state.user.id);
        await this.UserClient.Update(req);
        this.toggleModal();
      }
    } catch (err) {
      alert(err);
    }
  }

  toggleEditing() {
    this.setState(prevState => ({
      isEditing: !prevState.isEditing,
    }));
  }

  updateUser<K extends keyof User.AsObject>(prop: K) {
    return async (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
      const user = new User();
      const upperCaseProp = `${prop[0].toUpperCase()}${prop.slice(1)}`;
      const methodName = `set${upperCaseProp}`;
      user.setId(this.state.user.id);
      //@ts-ignore
      user[methodName](e.target.value);
      user.setFieldMaskList([upperCaseProp]);
      const updatedUser = await this.UserClient.Update(user);
      this.setState(() => ({ user: updatedUser }));
    };
  }
  updatePassword = this.updateUser('pwd');
  updateLogin = this.updateUser('login');
  updateEmail = this.updateUser('email');
  updateCellPhone = this.updateUser('cellphone');

  updateZipCode = this.updateUser('zip');
  updateCity = this.updateUser('city');
  updateStreetAddress = this.updateUser('address');
  updateFirstName = this.updateUser('firstname');
  updateLastName = this.updateUser('lastname');

  async componentDidMount() {
    await this.UserClient.GetToken('gavinorr', 'G@vin123');
    await this.fetchUser();
  }

  render() {
    const { isEditing } = this.state;
    if (this.state.user.id === 0) {
      return null;
    }
    return (
      <>
        <CssBaseline />
        <Grid
          style={{ paddingBottom: '20px' }}
          container
          alignItems="stretch"
          justify="center"
          direction="column"
        >
          <Grid container direction="row" justify="space-evenly">
            <TextField
              disabled={!isEditing}
              style={{ marginRight: '10px' }}
              defaultValue={this.state.user.firstname}
              onChange={this.updateFirstName}
              label={'First Name'}
            />
            <TextField
              disabled={!isEditing}
              defaultValue={this.state.user.lastname}
              onChange={this.updateLastName}
              label={'Last Name'}
            />
            <FormControlLabel
              control={
                <Switch
                  checked={this.state.isEditing}
                  onChange={this.toggleEditing}
                  value="isEditing"
                  color="primary"
                />
              }
              label={
                this.state.isEditing ? 'Editing Enabled' : 'Editing Disabled'
              }
            />
          </Grid>

          <Grid
            style={{ paddingBottom: '20px', paddingTop: '20px' }}
            container
            alignItems="stretch"
            justify="flex-start"
            direction="column"
          >
            <TextField
              style={{ paddingBottom: '10px', paddingTop: '10px' }}
              disabled={!isEditing}
              defaultValue={this.state.user.address}
              onChange={this.updateStreetAddress}
              label={'Street Address'}
            />
            <TextField
              style={{ paddingBottom: '10px', paddingTop: '10px' }}
              disabled={!isEditing}
              defaultValue={this.state.user.city}
              onChange={this.updateCity}
              label={'City'}
            />
            <TextField
              style={{ paddingBottom: '10px', paddingTop: '10px' }}
              disabled={!isEditing}
              defaultValue={this.state.user.zip}
              onChange={this.updateZipCode}
              label={'Zip Code'}
            />

            <TextField
              style={{ paddingBottom: '10px', paddingTop: '10px' }}
              disabled={!isEditing}
              defaultValue={this.state.user.cellphone}
              onChange={this.updateCellPhone}
              label={'Phone Number'}
            />
            <TextField
              style={{
                paddingBottom: '10px',
                paddingTop: '10px',
                paddingRight: 10,
              }}
              disabled
              defaultValue={this.state.user.email}
              onChange={this.updateEmail}
              label={'Email'}
            />
          </Grid>
          <Grid
            container
            direction="row"
            justify="space-evenly"
            style={{ marginTop: 10 }}
          >
            <Button
              disabled={!isEditing}
              onClick={this.toggleLoginModal}
              variant="contained"
              style={{ width: '30%' }}
            >
              Change Login
            </Button>
            <Button
              disabled={!isEditing}
              onClick={this.toggleModal}
              variant="contained"
              style={{ width: '30%' }}
            >
              Change Password
            </Button>
          </Grid>
        </Grid>
        <Modal
          open={this.state.isModalOpen}
          onClose={this.toggleModal}
          style={{
            margin: '10px',
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <Paper
            style={{
              padding: 20,
              width: '30%',
            }}
          >
            <Grid container direction={'column'}>
              <Grid
                container
                direction="row"
                justify="space-between"
                alignItems="center"
              >
                <Typography component="h2" variant="h6">
                  Change Password
                </Typography>
                <IconButton onClick={this.toggleModal}>
                  <CloseIcon />
                </IconButton>
              </Grid>
              <Divider />
              <TextField
                disabled={!isEditing}
                label={'Old Password'}
                type="password"
                inputProps={{ ref: this.oldPassword }}
              />
              <TextField
                disabled={!isEditing}
                label={'New Password'}
                type="password"
                inputProps={{
                  ref: this.newPassword,
                }}
              />
              <TextField
                disabled={!isEditing}
                label={'Repeat Password'}
                type="password"
                inputProps={{
                  ref: this.reTypePassword,
                }}
              />
              <Button
                endIcon={<SendIcon />}
                variant="contained"
                style={{ padding: 5, marginTop: 10 }}
                onClick={this.handleUpdatePassword}
              >
                Submit
              </Button>
            </Grid>
          </Paper>
        </Modal>
        <Modal
          open={this.state.isLoginModalOpen}
          onClose={this.toggleLoginModal}
          style={{
            margin: '10px',
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <Paper
            style={{
              padding: 20,
              width: '30%',
              display: 'flex',
              flexDirection: 'column',
              justifyContent: 'center',
              alignItems: 'center',
            }}
          >
            <Grid
              container
              direction="row"
              justify="space-between"
              alignItems="center"
            >
              <Typography component="h1" variant="h6">
                Change Login
              </Typography>
              <IconButton onClick={this.toggleLoginModal}>
                <CloseIcon />
              </IconButton>
            </Grid>
            <Divider />
            <TextField
              fullWidth
              defaultValue={this.state}
              disabled={!isEditing}
              label={'Old Login'}
              inputProps={{
                ref: this.oldLogin,
              }}
            />
            <TextField
              fullWidth
              disabled={!isEditing}
              label={'New Login'}
              inputProps={{
                ref: this.newLogin,
              }}
            />
            <TextField
              fullWidth
              disabled={!isEditing}
              label={' ReType Login'}
              inputProps={{
                ref: this.reTypeLogin,
              }}
            />
            <Button
              endIcon={<SendIcon />}
              variant="contained"
              style={{ padding: 5, marginTop: 10, width: '100%' }}
              onClick={this.handleUpdateLogin}
            >
              Confirm
            </Button>
          </Paper>
        </Modal>
      </>
    );
  }
}
