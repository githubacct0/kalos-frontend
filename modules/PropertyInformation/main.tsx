import React from 'react';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import customTheme from '../Theme/main';
import SectionBar from './components/SectionBar';
import { CustomerInformation } from './components/CustomerInformation';
import { PropertyInfo } from './components/PropertyInfo';

interface props {
  userID: number;
  propertyId: number;
}

interface state {}

export class PropertyInformation extends React.PureComponent<props, state> {
  render() {
    const { userID, propertyId } = this.props;
    return (
      <ThemeProvider theme={customTheme.lightTheme}>
        <SectionBar
          title="Customer Information"
          buttons={[
            {
              label: 'Calendar',
              url: `/index.cfm?action=admin:service.calendar&calendarAction=week&userIds=${userID}`,
            },
            {
              label: 'Call History',
              url: `/index.cfm?action=admin:customers.listPhoneCallLogs&code=customers&id=${userID}`,
            },
            {
              label: 'Tasks',
              url: `/index.cfm?action=admin:tasks.list&code=customers&id=${userID}`,
            },
            {
              label: 'Add Notification',
              onClick: () => {}, // TODO: implement onClick
            },
            {
              label: 'Edit Customer Information',
              onClick: () => {}, // TODO: implement onClick
            },
            {
              label: 'Delete Customer',
              onClick: () => {}, // TODO: implement onClick
            },
          ]}
        />
        <CustomerInformation {...this.props} />
        <SectionBar
          title="Property Information"
          buttons={[
            {
              label: 'Tasks',
              url: `/index.cfm?action=admin:tasks.list&code=properties&id=${propertyId}`,
            },
            {
              label: 'Add Notification',
              onClick: () => {}, // TODO: implement onClick
            },
            {
              label: 'Change Property',
              onClick: () => {}, // TODO: implement onClick
            },
            {
              label: 'Owner Details',
              url: `/index.cfm?action=admin:customers.details&user_id=${userID}`,
            },
          ]}
        />
        <PropertyInfo {...this.props} />
      </ThemeProvider>
    );
  }
}
