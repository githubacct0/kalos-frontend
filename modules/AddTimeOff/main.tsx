import React, { FC } from 'react';
import { PageWrapper, PageWrapperProps } from '../PageWrapper/main';
import { TimeOff, Props as TimeOffProps } from '../ComponentsLibrary/TimeOff';

export const AddTimeOff: FC<TimeOffProps & PageWrapperProps> = props => (
  <PageWrapper {...props} userID={props.loggedUserId} withHeader>
    <TimeOff {...props} />
  </PageWrapper>
);
