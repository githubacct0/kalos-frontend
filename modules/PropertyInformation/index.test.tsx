import React from 'react';
import renderer from 'react-test-renderer';
import { PropertyInformation } from './main';
test('PropertyInformation renders correctly', () => {
  const tree = renderer
    .create(
      <PropertyInformation
        userID={32}
        propertyId={6152}
        loggedUserId={101253}
        withHeader
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot(); // Rendered correctly
});
