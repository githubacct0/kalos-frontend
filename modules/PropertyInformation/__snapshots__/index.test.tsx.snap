// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyInformation renders correctly 1`] = `
<div
  className="PageWrapper"
>
  <div
    className="PageWrapperMenu"
  />
  <div
    className="PageWrapperContent"
    style={
      Object {
        "padding": "0rem",
      }
    }
  >
    <div
      className="CustomerInformation"
    >
      <div
        className="CustomerInformationCustomerInformation"
      >
        <div
          className="SectionBar collapsable sticky"
        >
          <div
            className="SectionBarHeaderWrapper"
          >
            <div
              className="SectionBarHeader"
            >
              <div
                className="SectionBarTitleWrapper collapsable"
              >
                <h5
                  className="MuiTypography-root SectionBarTitle MuiTypography-h5"
                  onClick={[Function]}
                >
                  Customer Information
                   
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </h5>
              </div>
            </div>
            <div
              className="SectionBarActions"
            >
              <div
                className="Actions"
              >
                <div
                  className="ActionsActions"
                >
                  <a
                    className="ButtonLink"
                    href="https://app.kalosflorida.com/index.cfm?action=admin:service.calendar&calendarAction=week&userIds=32"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Calendar
                      </span>
                    </button>
                  </a>
                  <a
                    className="ButtonLink"
                    href="https://app.kalosflorida.com/index.cfm?action=admin:customers.listPhoneCallLogs&code=customers&id=32"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Call History
                      </span>
                    </button>
                  </a>
                  <a
                    className="ButtonLink"
                    href="https://app.kalosflorida.com/index.cfm?action=admin:tasks.list&code=customers&id=32"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Tasks
                      </span>
                    </button>
                  </a>
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Add Notification
                    </span>
                  </button>
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Edit
                    </span>
                  </button>
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Delete
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="InfoTable"
        >
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Name
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Business Name
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Primary Phone
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Cell Phone
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Alternate Phone
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Fax
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Billing Address
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Email
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Billing Terms
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Customer Notes
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "50%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Internal Notes
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Groups
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="CustomerInformationAsidePanel"
      >
        <div
          className="SectionBar collapsable sticky"
        >
          <div
            className="SectionBarHeaderWrapper"
          >
            <div
              className="SectionBarHeader"
            >
              <div
                className="SectionBarTitleWrapper collapsable"
              >
                <h5
                  className="MuiTypography-root SectionBarTitle MuiTypography-h5"
                  onClick={[Function]}
                >
                  System Information
                   
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </h5>
              </div>
            </div>
            <div
              className="SectionBarActions"
            />
          </div>
        </div>
        <div
          className="InfoTable"
        >
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Created
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Last Login
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Login ID
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Wishes to receive promotional emails
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
        </div>
        <div
          className="CustomerInformationPendingBilling SectionBar collapsable sticky"
        >
          <div
            className="SectionBarHeaderWrapper"
          >
            <div
              className="SectionBarHeader"
            >
              <div
                className="SectionBarTitleWrapper collapsable"
              >
                <h5
                  className="MuiTypography-root SectionBarTitle MuiTypography-h5"
                  onClick={[Function]}
                >
                  Pending Billing
                   
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </h5>
              </div>
            </div>
            <div
              className="SectionBarActions"
            >
              <div
                className="Actions"
              >
                <div
                  className="ActionsActions"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton size-small status-undefined variant-contained disabled MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
                    disabled={true}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      None
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="InfoTable"
        >
          <div
            className="InfoTableRow hoverable"
          >
            <div
              className="MuiTypography-root InfoTableItem MuiTypography-body1"
              style={
                Object {
                  "cursor": "default",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            >
              <strong
                className="InfoTableLabel"
              >
                Record Count
                : 
              </strong>
              <span
                className="InfoTableFake"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="SectionBar collapsable sticky"
    >
      <div
        className="SectionBarHeaderWrapper"
      >
        <div
          className="SectionBarHeader"
        >
          <div
            className="SectionBarTitleWrapper collapsable"
          >
            <h5
              className="MuiTypography-root SectionBarTitle MuiTypography-h5"
              onClick={[Function]}
            >
              Property Information
               
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </h5>
          </div>
        </div>
        <div
          className="SectionBarActions"
        />
      </div>
    </div>
    <div
      className="InfoTable"
    >
      <div
        className="InfoTableRow hoverable"
      >
        <p
          className="MuiTypography-root InfoTableNoEntries MuiTypography-body1"
        >
          No entries found.
        </p>
      </div>
    </div>
  </div>
</div>
`;
